core/metamodel/src/main/java/org/apache/isis/core/metamodel/adapter/ObjectAdapter.java

```diff
@@ -482,28 +469,36 @@ private Functions(){}
             return Util::unwrap;
         }
 
-        public static Function<Object, ObjectAdapter> adapterForUsing(final ObjectAdapterProvider adapterProvider) {
-            return adapterProvider::adapterFor;
-        }
-
-        @Deprecated
-        public static com.google.common.base.Function<Object, ObjectAdapter> adapter_ForUsing(final ObjectAdapterProvider adapterProvider) {
-            return new com.google.common.base.Function<Object, ObjectAdapter>() {
-                @Override
-                public ObjectAdapter apply(final Object pojo) {
-                    return adapterProvider.adapterFor(pojo);
-                }
-            };
-        }
     }
 
     /**
      * 
-     * @param persistedRootOid
+     * @param newOid
```

Comment:
could you add some comments as to why we (still) need to in effect update the RootOid of an adapter.

My guess is that this is to do with the "disposableAdapter" idea - can only call this method on a disposable adapter in order to clone it and create a "real" adapter?

If that's the case, then maybe the direction of travel is that eventually the "disposable adapter" and "regular adapters" will separate into two separate subtypes of a common supertype.  The common supertype will provide access to specifications etc, the "disposable adapter" has this "withOid(..)" as a factor for "regular adapters", and the "getOid()" method is only exposed by the regular adapters.

Reply:
It's not related to the disposable OA idea, I guess. 

There is one remaining reference to changing the pojo in an OA which is from the PersistenceSession implementation, when calling `objectAdapterContext.remapAsPersistent(adapter, persistentOid, this);` in ...

```java
/**
     * Called either when an entity is initially persisted, or when an entity is updated;
     * fires the appropriate lifecycle callback
     *
     * <p>
     * The implementation therefore uses Isis' {@link org.apache.isis.core.metamodel.adapter.oid.Oid#isTransient() oid}
     * to determine which callback to fire.
     */
    @Override
    public void enlistCreatedAndRemapIfRequiredThenInvokeIsisInvokePersistingOrUpdatedCallback(final Persistable pojo) {
        final ObjectAdapter adapter = adapterFor(pojo);

        final RootOid rootOid = (RootOid) adapter.getOid(); // ok since this is for a Persistable

        if (rootOid.isTransient()) {
            // persisting
            final RootOid persistentOid = objectAdapterContext.createPersistentOrViewModelOid(pojo);

            objectAdapterContext.remapAsPersistent(adapter, persistentOid, this);

            CallbackFacet.Util.callCallback(adapter, PersistedCallbackFacet.class);
            postLifecycleEventIfRequired(adapter, PersistedLifecycleEventFacet.class);

            changedObjectsServiceInternal.enlistCreated(adapter);

        } else {
            // updating;
            // the callback and transaction.enlist are done in the preDirty callback
            // (can't be done here, as the enlist requires to capture the 'before' values)
            CallbackFacet.Util.callCallback(adapter, UpdatedCallbackFacet.class);
            postLifecycleEventIfRequired(adapter, UpdatedLifecycleEventFacet.class);
        }

        Version versionIfAny = getVersionIfAny(pojo);
        adapter.setVersion(versionIfAny);
    }
```

Reply:
regarding the disposable OA idea: we might just uses `Instance` instead of OA which also privides the ObjectSpecification. but needs further investigation